packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -709,259 +818,180 @@ public void resumeVideoRecording(@NonNull final Result result) {
     result.success(null);
   }
 
-  public void setFlashMode(@NonNull final Result result, FlashMode mode)
-      throws CameraAccessException {
-    // Get the flash availability
-    Boolean flashAvailable =
-        cameraManager
-            .getCameraCharacteristics(cameraDevice.getId())
-            .get(CameraCharacteristics.FLASH_INFO_AVAILABLE);
-
-    // Check if flash is available.
-    if (flashAvailable == null || !flashAvailable) {
-      result.error("setFlashModeFailed", "Device does not have flash capabilities", null);
-      return;
-    }
-
-    // If switching directly from torch to auto or on, make sure we turn off the torch.
-    if (flashMode == FlashMode.torch && mode != FlashMode.torch && mode != FlashMode.off) {
-      updateFlash(FlashMode.off);
+  /**
+   * Dart handler when it's time to set a new flash mode. This will try to set a new flash mode to
+   * the current camera.
+   *
+   * @param result Flutter result.
+   * @param newMode new mode.
+   */
+  public void setFlashMode(@NonNull final Result result, FlashMode newMode) {
```

Comment:
is newMode Nullable or NonNull?  why is the annotation in one argument, but not the other?